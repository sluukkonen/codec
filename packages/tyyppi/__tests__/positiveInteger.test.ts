import { positiveInteger } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should accept positive integers", () => {
  expectParseSuccess(positiveInteger, 1)
  expectParseSuccess(positiveInteger, 2)
  expectParseSuccess(positiveInteger, 3)
})

test("should not accept 0", () => {
  expectParseFailure(positiveInteger, 0)
})

test("should not accept negative integers", () => {
  expectParseFailure(positiveInteger, -1)
  expectParseFailure(positiveInteger, -2)
  expectParseFailure(positiveInteger, -2)
})

test("should not accept non-integers", () => {
  expectParseFailure(positiveInteger, -0.5)
  expectParseFailure(positiveInteger, 0.5)
})

test("should not accept NaN", () => {
  expectParseFailure(positiveInteger, NaN)
})
