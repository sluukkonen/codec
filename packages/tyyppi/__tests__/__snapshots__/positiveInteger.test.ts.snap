// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not accept 0 1`] = `
[
  {
    "code": "too_small",
    "message": "Expected the number to be greater than or equal to 1",
    "min": 1,
    "path": [],
    "type": "number",
  },
]
`;

exports[`should not accept NaN 1`] = `
[
  {
    "code": "is_nan",
    "message": "Expected a number, received NaN",
    "path": [],
  },
]
`;

exports[`should not accept negative integers 1`] = `
[
  {
    "code": "too_small",
    "message": "Expected the number to be greater than or equal to 1",
    "min": 1,
    "path": [],
    "type": "number",
  },
]
`;

exports[`should not accept negative integers 2`] = `
[
  {
    "code": "too_small",
    "message": "Expected the number to be greater than or equal to 1",
    "min": 1,
    "path": [],
    "type": "number",
  },
]
`;

exports[`should not accept negative integers 3`] = `
[
  {
    "code": "too_small",
    "message": "Expected the number to be greater than or equal to 1",
    "min": 1,
    "path": [],
    "type": "number",
  },
]
`;

exports[`should not accept non-integers 1`] = `
[
  {
    "code": "invalid_integer",
    "message": "Expected the number to be an integer",
    "path": [],
  },
]
`;

exports[`should not accept non-integers 2`] = `
[
  {
    "code": "invalid_integer",
    "message": "Expected the number to be an integer",
    "path": [],
  },
]
`;
